internal/kafka/internal/handlers/kafka.go

```diff
@@ -65,6 +67,11 @@ func (h kafkaHandler) Create(w http.ResponseWriter, r *http.Request) {
 
 			convKafka.CloudProvider, convKafka.Region, _ = getCloudProviderAndRegion(ctx, h.service, &kafkaRequestPayload, h.providerConfig)
 
+			// enterprise kafkas should be assigned to specified cluster, if its ID is provided
+			if !shared.StringEmpty(kafkaRequestPayload.ClusterId) {
+				convKafka.ClusterID = *kafkaRequestPayload.ClusterId
```

Comment:
@pawelpaszki 

What I am about to say is something that's not noted in the JIRA ask but it is something we've to do.

At the moment, we've the kafka unassignment logic https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/internal/kafka/internal/services/data_plane_kafka.go#L396 

The intention was that, if a Kafka is rejected in the data plane due to insufficient resources, we reassign it onto a different data plane. 
This logic has to be limited only to managed kafkas and not applied for enterprise kafkas.
My proposal is then to add a check on https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ba3f5f14f64443a12a20a2df6d38e045b3637054/internal/kafka/internal/services/data_plane_kafka.go#L396 that if the kafka billing model is `enterprise` then it is not re-assigned i.e we keep it on the existing cluster.

Now the question becomes, what should we do if a kafka is rejected due to insufficient resources?
1. We could keep them to provisioning state and log an error message to sentry. This will be preferably, until we  know what we do under this case. 
2. change the state of the kafka to `failed` and add `cluster full` as failed_reason. This could be good for user as they see the failures reason

Reply:
I will go with option (1)